module_utils/network_lsr/ethtool.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+import array
+import binascii
+import struct
+import fcntl
+import socket
+
+
+ETHTOOL_GPERMADDR = 0x00000020
+SIOCETHTOOL = 0x8946
+
+
+def get_perm_addr(ifname):
+    """
+    Return the Permanent address value for the specified interface using the
+    ETHTOOL_GPERMADDR ioctl command. The address is returned as string in
+    python2 so it is needed to convert it to a bytearray and convert it again
+    to a hex string again.
+
+    Please for further documentation, see:
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L734
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L1388
+    https://git.kernel.org/pub/scm/network/ethtool/ethtool.git/tree/ethtool.c#n4172
+    """
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sockfd = sock.fileno()
+        ifname = ifname.encode("utf-8")
+        ecmd = array.array("B", struct.pack("II6s", ETHTOOL_GPERMADDR, 6, b"\x00" * 6))
+        ifreq = struct.pack("16sP", ifname, ecmd.buffer_info()[0])
+
+        fcntl.ioctl(sockfd, SIOCETHTOOL, ifreq)
+        res = ecmd.tostring()
```

Comment:
will be removed in Python 3.9: https://docs.python.org/3/library/array.html#array.array.tostring, OTOH `tobytes` does not seem to be available in 2.7.

Reply:
I am using `tostring` because is the way to make this on python2 and python3. As python is now deprecated, it means they are not going to make them compatible in future versions. This will be a problem for projects that are currently supporting python2 and python3.

There is no solution for this, IMO.